ush/python/pygfs/task/nexus_emission.py

```diff
@@ -0,0 +1,459 @@
+        - It will log the processing steps and any issues encountered.
+        Raises
+        ------
+        WorkflowException
+            If the NEXUS preprocessor executable is not found
+            If the working directory does not exist
+            If no emission files are found for processing
+        """
+        logger.info(f"Running NEXUS emission preprocessor in {self.task_config.DATA}")
+        logger.info(f"NEXUS Logs: {self.task_config.DATA}/stdout")
+        logger.info(f"NEXUS Logs: {self.task_config.DATA}/stderr")
+        logger.info(f"NEXUS Logs: {self.task_config.DATA}/NEXUS.log")
+
+        if not os.path.exists(self.task_config.DATA):
+            raise WorkflowException(f"Working directory does not exist: {self.task_config.DATA}")
+
+        exe = Executable(self.task_config.APRUN)
+
+        if os.path.exists("nexus.x") is False:
+            raise WorkflowException("NEXUS preprocessor executable 'nexus.x' not found in PATH")
+
+        arg_list = ['./nexus.x', '-c', self.task_config.NEXUS_CONFIG_NAME]
+        exe(*arg_list, output='stdout', error='stderr')
+
+        logger.info("Concatenating processed NEXUS files...")
+
+        # sort the files even though they should be sorted already | safety check
+        files = sorted(self.processed_nexus_files)
+
+        for i in files:
+
+            if not os.path.exists(i):
+                logger.warning(f"NEXUS file not found: {i}")
+                continue
+            else:
+                logger.info(f"NEXUS file found: {i}")
+
+        for f, dates in zip(files, self.forecast_dates):
+            logger.info(f" - {f}, {dates}")
+
+        # find the day indexes for each unique day
+        # this returns a dictionary
+        # example:
+        # {
+        #     datetime.date(2024, 1, 5): [0, 1, 3],
+        #     datetime.date(2024, 1, 6): [2]
+        # }
+        day_indexes = _get_day_indices(self.forecast_dates[:-1])  # hemco doesn't write out the last timestep
+        # now loop over each days
+        for date, indexes in day_indexes.items():
+            day_str = date.strftime('%Y%m%d')
+            logger.info(f"Processing NEXUS files for date: {date}")
+
+            dsets = []
+            for index in indexes:
+                # list files for log
+                logger.info(f" - {files[index]}, {index}")
+
+                # now concatenate the files per day
+                if os.path.exists(files[index]) is False:
+                    break
+                ds = xr.open_dataset(files[index], decode_cf=False)
+
+                # update time coordinate
+                ds = ds.assign_coords(time=('time', [index]))
+
+                # set time units to reference start-date
+                ds.time.attrs['units'] = self.start_date.strftime('hours since %Y-%m-%d %H:00:00')
+
+                # append
+                dsets.append(ds)
+
+            # concatenate all the files for this day
+            if dsets is None:
```

Comment:
This won't ever be true as `dsets` is initialized to `[]`.

```suggestion
            if len(dsets) == 0:
```